Sources/UIKit/AdaptiveStyle.swift

```diff
@@ -177,23 +194,41 @@ extension AdaptiveStyle: EmbeddedTransformation {
             return [EmbeddedTransformationHelpers.Key.type: Value.body]
         case .preferred:
             return [EmbeddedTransformationHelpers.Key.type: Value.preferred]
+        case .fontMetrics(let textStyle, let maxPointSize):
+            var attributes: StyleAttributes = [
+                EmbeddedTransformationHelpers.Key.type: Value.fontMetrics,
+                EmbeddedTransformationHelpers.Key.textStyle: textStyle,
+            ]
+            if let maxPointSize = maxPointSize {
+                attributes[EmbeddedTransformationHelpers.Key.maxPointSize] = maxPointSize
+            }
+            return attributes
         }
     }
 
     static func from(dictionary dict: StyleAttributes) -> EmbeddedTransformation? {
         switch (dict[EmbeddedTransformationHelpers.Key.type] as? String,
                 dict[EmbeddedTransformationHelpers.Key.size] as? CGFloat,
-                dict[Key.fontName] as? String) {
-        case (Value.control?, nil, nil):
+                dict[Key.fontName] as? String,
+                dict[EmbeddedTransformationHelpers.Key.textStyle] as? BonMotTextStyle,
+                dict[EmbeddedTransformationHelpers.Key.maxPointSize] as? CGFloat) {
+        case (Value.control?, nil, nil, nil, nil):
```

Comment:
I really love this switch statement syntax but it feels kind of clunky with a 5 value tuple (previously there were only 3). I wasn't sure how best to refactor but I'm open to suggestions if reviewers agree it should be rethought.